saleor/graphql/shop/mutations.py

```diff
@@ -288,3 +288,125 @@ def perform_mutation(cls, _root, info, key_type):
 
         instance.delete()
         return AuthorizationKeyDelete(authorization_key=instance, shop=Shop())
+
+
+class StaffNotificationInput(graphene.InputObjectType):
+    staff_email = graphene.String(
+        required=False,
+        description="Staff email (instead of providinf staff member user).",
+    )
+    active = graphene.Boolean(required=False, description="Client secret.")
+    user = graphene.ID(required=False, description="The ID of the user to assign.")
+
+
+class StaffNotificationAdd(BaseMutation):
+    staff_notification = graphene.Field(
+        StaffNotification, description="StaffNotification."
+    )
+
+    class Arguments:
+        input = StaffNotificationInput(
+            required=True, description="Fields required to update a product."
+        )
+
+    class Meta:
+        description = "Adds an staff notification."
+        permissions = ("site.manage_settings",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        input_data = data["input"]
+
+        if "staff_email" not in input_data and "user" not in input_data:
+            raise ValidationError(
+                {
+                    "staff_notification": ValidationError(
+                        "User or staff email is required", code=ShopErrorCode.REQUIRED
+                    )
+                }
+            )
+        staff_notification_kwargs = {
+            "site_settings": info.context.site.settings,
+            "active": input_data.get("active", True),
+        }
+        if "user" in input_data:
+            user = cls.get_node_or_error(info, input_data["user"], only_type=User)
+            staff_notification_kwargs["user"] = user
+        elif "staff_email" in input_data:
+            staff_email = input_data["staff_email"]
+            staff_user = account_models.User.objects.filter(email=staff_email)
+            if staff_user.exists():
+                staff_notification_kwargs["user"] = staff_user.first()
```

Comment:
It could be simplified a little
```suggestion
            staff_user = account_models.User.objects.filter(email=staff_email).first()
            if staff_user:
                staff_notification_kwargs["user"] = staff_user
```